test/denormalized_has_many_test.rb

```diff
@@ -12,7 +12,23 @@ def setup
     @record.reload
   end
 
+  def test_uncached_record_from_the_db_will_use_normal_association_as_ary
+    # Test configuration where fetch association returns an array
+    IdentityCache.allow_fetch_to_return_array do    
+      assert_equal IdentityCache.fetch_returns_relation, false
+      expected = @record.associated_records.to_a
+      record_from_db = Item.find(@record.id)
+      assert_queries(1) do
+        assert_memcache_operations(0) do
+          assert_equal expected, record_from_db.fetch_associated_records
```

Comment:
Actually, it looks like the denormalized and normalized test_fetch_association_does_not_allow_chaining_with_ary tests already provide regression tests.  What else are you trying to test with these other tests?